server/aws-lsp-codewhisperer/src/language-server/languageDetection.ts

```diff
@@ -1,58 +1,96 @@
 import { TextDocument } from '@aws/language-server-runtimes/server-interface'
 
 export type CodewhispererLanguage =
-    | 'java'
-    | 'python'
-    | 'jsx'
-    | 'javascript'
-    | 'typescript'
-    | 'tsx'
-    | 'csharp'
     | 'c'
     | 'cpp'
-    | 'cpp'
+    | 'csharp'
     | 'go'
+    | 'java'
+    | 'javascript'
+    | 'jsx'
     | 'kotlin'
     | 'php'
+    | 'plaintext'
+    | 'python'
     | 'ruby'
     | 'rust'
     | 'scala'
     | 'shell'
-    | 'shell'
     | 'sql'
-    | 'plaintext'
+    | 'tsx'
+    | 'typescript'
+
+    // These work as well
+    | 'json'
+    | 'tf'
+    | 'vue'
+    | 'yaml'
 
 // This will be extended as more language features
 // are integrated into the language server and clients.
 // See: https://microsoft.github.io/language-server-protocol/specifications/lsp/3.18/specification/#textDocumentItem
-const supportedFileTypes: CodewhispererLanguage[] = ['c', 'cpp', 'csharp', 'javascript', 'python', 'typescript']
+const supportedFileTypes: CodewhispererLanguage[] = ['c', 'cpp', 'csharp', 'javascript', 'jsx', 'python', 'typescript']
+
 export const supportedSecurityScanLanguages: CodewhispererLanguage[] = ['csharp']
-const supportedExtensions: { [key: string]: CodewhispererLanguage } = {
+
+export const languageByExtension: { [key: string]: CodewhispererLanguage } = {
     '.c': 'c',
-    '.h': 'c',
     '.cpp': 'cpp',
-    '.hpp': 'cpp',
     '.cs': 'csharp',
+    '.h': 'c',
+    '.hpp': 'cpp',
+    '.go': 'go',
+    '.kt': 'kotlin',
+    '.kts': 'kotlin',
+    '.java': 'java',
     '.js': 'javascript',
+    '.json': 'json',
+    '.jsx': 'jsx',
+    '.php': 'php',
     '.py': 'python',
+    '.rb': 'ruby',
+    '.rs': 'rust',
+    '.sc': 'scala',
+    '.scala': 'scala',
+    '.sh': 'shell',
+    '.sql': 'sql',
+    '.tf': 'tf',
     '.ts': 'typescript',
+    '.tsx': 'tsx',
+    '.vue': 'vue',
+    '.yaml': 'yaml',
+    '.yml': 'yaml',
+}
+
+export const additionalLanguageMapping: { [key: string]: CodewhispererLanguage } = {
```

Comment:
Where do we get these language identifiers from? If it's not `getCodeWhispererLanguageIdByTextDocumentLanguageId`, `getCodeWhispererLanguageIdByExtension`, what is the source of these identifiers on the left side of this map?

Reply:
This mapping is `{ [textDocumentLanguageId: string]: CWPRLanguageId }`. I noticed `getLanguageId` is not checking against the entire list so I decided to create a single map that has all the mapping.